atc/db/volume.go

```diff
@@ -371,13 +371,20 @@ func (volume *createdVolume) initializeResourceCache(resourceCache UsedResourceC
 
 	defer tx.Rollback()
 
-	workerResourceCache, err := WorkerResourceCache{
+	workerResourceCache, valid, err := WorkerResourceCache{
 		WorkerName:    volume.WorkerName(),
 		ResourceCache: resourceCache,
 	}.FindOrCreate(tx, sourceWorkerName)
 	if err != nil {
 		return err
 	}
+	if !valid {
```

Comment:
Good point. To be honest, I'm not sure exactly what setting `team_id` to `nil` does, but it's probably safest for us to do that here as well I suppose

Reply:
Uncommenting the `Set("team_id", nil)` on the original update doesn't seem to cause any tests to fail - I wonder if it's just legacy logic or if it's still important but untested 🤔

Reply:
Actually, thinking about it more, I think it probably doesn't make sense to set `team_id` to `nil` - I suspect it was only set to `nil` since even if a resource cache volume was produced on a workload for team X, it can still be used by a workload for team Y. But since this case is for when the volume will *not* become the resource cache, we shouldn't need to do this.

In the event of a unique violation below: https://github.com/concourse/concourse/blob/b45990d3a4122e357b19b9d9ddb58590e3ac95c0/atc/db/volume.go#L389-L403

...we still don't update `team_id` to `nil`, and this really represents a similar case - another volume "won the race" to being the resource cache.

Reply:
Ohhh yeah I see, this should just not work if there was a dup anyways because of the unique index on `worker_resource_cache_id`. Cool thanks for digging into it for me :D